### ISSUE TITLE
Error response from daemon: server error from portlayer : [DELETE /containers/{id}][500] containerRemoveInternalServerError

### ISSUE THREAD

**Issue by mhagen-vmware** [vmware/vic#1823]
Looks like failed to stop the container properly before trying to remove failed:
[integration_test_logs_3662_1470347704.zip](https://github.com/vmware/vic/files/402800/integration_test_logs_3662_1470347704.zip)

```
time="2016-08-04T21:53:51Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.(*Container).stop] Container.stop" 
time="2016-08-04T21:53:51Z" level=error msg="HandshakeClient: failed to read expected SYN-ACK: n=2, err=read tcp 192.168.31.148:8080->192.168.31.12:10735: i/o timeout buf=[0x1515]" 
time="2016-08-04T21:53:52Z" level=error msg="HandshakeClient: failed to read expected SYN-ACK: n=1, err=read tcp 192.168.31.148:8080->192.168.31.12:10735: i/o timeout buf=[0x15]" 
time="2016-08-04T21:53:52Z" level=error msg="HandshakeClient: failed to read expected SYN-ACK: n=1, err=read tcp 192.168.31.148:8080->192.168.31.12:10735: i/o timeout buf=[0x15]" 
time="2016-08-04T21:53:52Z" level=debug msg="HandshakeClient: did not receive synack: 0x0697 != 0x1506" 
time="2016-08-04T21:53:52Z" level=debug msg="discarding following 1 bytes from input channel\n" 
time="2016-08-04T21:53:52Z" level=debug msg="[48]\n" 
time="2016-08-04T21:53:52Z" level=debug msg="caught EOF" 
time="2016-08-04T21:53:52Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/portlayer/exec.(*Container).stop] [698.372953ms] Container.stop" 
time="2016-08-04T21:53:52Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/portlayer/exec.(*Container).Commit] [901.171181ms] Committing handle" 
time="2016-08-04T21:53:52Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).CommitHandler] [902.095679ms] Containers.CommitHandler" 
time="2016-08-04T21:53:52Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerListHandler] Containers.GetContainerListHandler" 
time="2016-08-04T21:53:52Z" level=debug msg="Skipping non-container vm VCH-3662-5081" 
time="2016-08-04T21:53:52Z" level=debug msg="Skipping poweredOff VM silly_snyder-d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a" 
time="2016-08-04T21:53:52Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerListHandler] [17.346009ms] Containers.GetContainerListHandler" 
time="2016-08-04T21:53:52Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerListHandler] Containers.GetContainerListHandler" 
time="2016-08-04T21:53:52Z" level=debug msg="Skipping non-container vm VCH-3662-5081" 
time="2016-08-04T21:53:52Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).GetContainerListHandler] [29.366292ms] Containers.GetContainerListHandler" 
time="2016-08-04T21:53:52Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).RemoveContainerHandler] Containers.RemoveContainerHandler" 
time="2016-08-04T21:53:52Z" level=debug msg="[BEGIN] [github.com/vmware/vic/lib/portlayer/exec.(*Container).Remove] Container.Remove" 
time="2016-08-04T21:53:53Z" level=error msg="Failed to invoke operation: The operation is not allowed in the current state." 
time="2016-08-04T21:53:53Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/portlayer/exec.(*Container).Remove] [24.278848ms] Container.Remove" 
time="2016-08-04T21:53:53Z" level=debug msg="[ END ] [github.com/vmware/vic/lib/apiservers/portlayer/restapi/handlers.(*ContainersHandlersImpl).RemoveContainerHandler] [24.776016ms] Containers.RemoveContainerHandler" 
```

**Comment by hickeng** [comment]
The full portion of the relevant hostd.log fragment is:
```
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] SOCKET 1492 (39)
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] recv detected client closed connection
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] VigorTransportClientProcessError: Remote connection failure
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] VigorTransportClientDrainRecv: draining read.
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] SOCKET 1492 (39)
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] recv detected client closed connection
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] VigorTransportClientProcessError: closing connection.
2016-08-04T21:53:53.006Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] VigorTransportClientManageConnection: connection closed.
2016-08-04T21:53:53.030Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] CnxAuthdProtoSecureConnect: Unencrypted connection, skipping thumbprint exchange.
2016-08-04T21:53:53.048Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] CnxConnectAuthd: Returning false because CnxAuthdProtoConnect failed
2016-08-04T21:53:53.048Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] Cnx_Connect: Returning false because CnxConnectAuthd failed
2016-08-04T21:53:53.048Z info hostd[FFCB2B70] [Originator@6876 sub=vm:Cnx_Connect: Error message: There is no VMware process running for config file /vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx]
2016-08-04T21:53:53.048Z warning hostd[FFCB2B70] [Originator@6876 sub=vm:VigorTransportClientManageConnection: Failed to re-connect to VM /vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx]
2016-08-04T21:53:53.052Z info hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] Error encountered while retrieving configuration. Marking configuration as invalid: N3Vim5Fault20GenericVmConfigFault9ExceptionE(vim.fault.GenericVmConfigFault)
2016-08-04T21:53:53.053Z warning hostd[83B44B70] [Originator@6876 sub=Hostsvc] Unable to get worldId for cartel 1242611 : Sysinfo error on operation returned status : No cartel by that name. Please see the VMkernel log for detailed error information
2016-08-04T21:53:53.053Z warning hostd[83B44B70] [Originator@6876 sub=Vigor.Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] Could not retrieve offline Bootstrap state: N3Vim5Fault20GenericVmConfigFault9ExceptionE(vim.fault.GenericVmConfigFault)
2016-08-04T21:53:53.053Z error hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] Could not fetch Vigor vmx state: N3Vim5Fault20GenericVmConfigFault9ExceptionE(vim.fault.GenericVmConfigFault)
2016-08-04T21:53:53.053Z warning hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] Unstable power state, assuming off
2016-08-04T21:53:53.053Z info hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] State Transition (VM_STATE_ON_SHUTTING_DOWN -> VM_STATE_OFF)
2016-08-04T21:53:53.054Z warning hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] Failed to find activation record, event user unknown.
2016-08-04T21:53:53.054Z info hostd[83B44B70] [Originator@6876 sub=Vimsvc.ha-eventmgr] Event 34301 : silly_snyder-d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a on  vic-esx-08.ci.drone.local in ha-datacenter is powered off
2016-08-04T21:53:53.054Z error hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] Could not fetch Vigor vmx state: N3Vim5Fault20GenericVmConfigFault9ExceptionE(vim.fault.GenericVmConfigFault)
2016-08-04T21:53:53.054Z warning hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx] Unstable power state, assuming off
2016-08-04T21:53:53.054Z info hostd[809C1B70] [Originator@6876 sub=Guestsvc.GuestFileTransferImpl] Entered VmPowerStateListener
2016-08-04T21:53:53.054Z info hostd[809C1B70] [Originator@6876 sub=Guestsvc.GuestFileTransferImpl] VmPowerStateListener succeeded
2016-08-04T21:53:53.054Z info hostd[809C1B70] [Originator@6876 sub=Hbrsvc] Replicator: powerstate change VM: 2068 Old: 1 New: 0
2016-08-04T21:53:53.054Z info hostd[809C1B70] [Originator@6876 sub=Hbrsvc] Replicator: Poweroff for VM: (id=2068)
2016-08-04T21:53:53.059Z info hostd[83D8BB70] [Originator@6876 sub=Vimsvc.TaskManager opID=030cbbf7 user=root] Task Created : haTask-2068-vim.VirtualMachine.reconfigure-368307
2016-08-04T21:53:53.060Z info hostd[83E0CB70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbbf7 user=root] State Transition (VM_STATE_OFF -> VM_STATE_RECONFIGURING)
2016-08-04T21:53:53.060Z error hostd[83E0CB70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbbf7 user=root] Could not fetch Vigor vmx state: N3Vim5Fault20GenericVmConfigFault9ExceptionE(vim.fault.GenericVmConfigFault)
2016-08-04T21:53:53.060Z warning hostd[83E0CB70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbbf7 user=root] Unstable power state, assuming off
2016-08-04T21:53:53.064Z info hostd[FFCB2B70] [Originator@6876 sub=Libs] VigorOnlineDisconnectCb: connection closed (is final).
2016-08-04T21:53:53.206Z info hostd[83E0CB70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbbf7 user=root] State Transition (VM_STATE_RECONFIGURING -> VM_STATE_OFF)
2016-08-04T21:53:53.207Z info hostd[83E0CB70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbbf7 user=root] Marking VirtualMachine invalid
2016-08-04T21:53:53.207Z info hostd[83E0CB70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbbf7 user=root] State Transition (VM_STATE_OFF -> VM_STATE_INVALID_CONFIG)
2016-08-04T21:53:53.207Z info hostd[83E0CB70] [Originator@6876 sub=Vimsvc.ha-eventmgr opID=030cbbf7 user=root] Event 34302 : Reconfigured silly_snyder-d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a on vic-esx-08.ci.drone.local in ha-datacenter
2016-08-04T21:53:53.207Z info hostd[83E0CB70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbbf7 user=root] Send config update invoked
2016-08-04T21:53:53.255Z info hostd[83E0CB70] [Originator@6876 sub=Vimsvc.TaskManager opID=030cbbf7 user=root] Task Completed : haTask-2068-vim.VirtualMachine.reconfigure-368307 Status success
2016-08-04T21:53:53.408Z info hostd[83B44B70] [Originator@6876 sub=Vimsvc.TaskManager opID=030cbc0b user=root] Task Created : haTask-2068-vim.VirtualMachine.reconfigure-368315
2016-08-04T21:53:53.409Z warning hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbc0b user=root] State Transition (VM_STATE_INVALID_CONFIG -> VM_STATE_RECONFIGURING) not allowed for this Vm
2016-08-04T21:53:53.409Z warning hostd[83B44B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbc0b user=root] Failed to start Reconfiguring: vim.fault.InvalidState
2016-08-04T21:53:53.409Z info hostd[83B44B70] [Originator@6876 sub=Default opID=030cbc0b user=root] AdapterServer caught exception: vim.fault.InvalidState
2016-08-04T21:53:53.409Z info hostd[83B44B70] [Originator@6876 sub=Vimsvc.TaskManager opID=030cbc0b user=root] Task Completed : haTask-2068-vim.VirtualMachine.reconfigure-368315 Status error
2016-08-04T21:53:55.849Z info hostd[82A40B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/VCH-3662-5081/VCH-3662-5081.vmx] Send config update invoked
pam_unix(vmware-authd:auth): authentication failure; logname= uid=0 euid=0 tty= ruser= rhost=  user=root
[LikewiseGetDomainJoinInfo:355] QueryInformation(): ERROR_FILE_NOT_FOUND (2/0):
Rejected password for user root from 192.168.31.148
2016-08-04T21:54:05.742Z info hostd[82A40B70] [Originator@6876 sub=Vimsvc.ha-eventmgr opID=030cbc15] Event 34303 : Cannot login root@192.168.31.148
2016-08-04T21:54:08.744Z info hostd[82A40B70] [Originator@6876 sub=Solo.Vmomi] Activation [N5Vmomi10ActivationE:0x82005370] : Invoke done [login] on [vim.SessionManager:ha-sessionmgr]
2016-08-04T21:54:08.745Z verbose hostd[82A40B70] [Originator@6876 sub=Solo.Vmomi] Arg userName:
--> "root"
2016-08-04T21:54:08.745Z verbose hostd[82A40B70] [Originator@6876 sub=Solo.Vmomi] Arg password:
--> (not shown)
-->
2016-08-04T21:54:08.745Z verbose hostd[82A40B70] [Originator@6876 sub=Solo.Vmomi] Arg locale:
--> (null)
2016-08-04T21:54:08.745Z info hostd[82A40B70] [Originator@6876 sub=Solo.Vmomi] Throw vim.fault.InvalidLogin
2016-08-04T21:54:08.745Z info hostd[82A40B70] [Originator@6876 sub=Solo.Vmomi] Result:
--> (vim.fault.InvalidLogin) {
-->    faultCause = (vmodl.MethodFault) null,
-->    msg = ""
--> }
[LikewiseGetDomainJoinInfo:355] QueryInformation(): ERROR_FILE_NOT_FOUND (2/0):
Accepted password for user root from 192.168.31.107
2016-08-04T21:54:09.323Z info hostd[83E0CB70] [Originator@6876 sub=Vimsvc.ha-eventmgr opID=030cbc17] Event 34304 : User root@192.168.31.107 logged in as Go-http-client/1.1
2016-08-04T21:54:09.510Z info hostd[82A40B70] [Originator@6876 sub=Hostsvc.VsanSystemProvider opID=030cbc35 user=root] Complete, runtime info: (vim.vsan.host.VsanRuntimeInfo) {
-->    accessGenNo = 0,
--> }
2016-08-04T21:54:09.868Z info hostd[803D5B70] [Originator@6876 sub=Vimsvc.TaskManager opID=030cbc40 user=root] Task Created : haTask-2068-vim.ManagedEntity.destroy-368322
2016-08-04T21:54:09.869Z info hostd[809C1B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbc40 user=root] Destroy VM called
2016-08-04T21:54:09.869Z error hostd[809C1B70] [Originator@6876 sub=Vmsvc.vm:/vmfs/volumes/5790b7d6-e2847f28-db9f-0050569c86f0/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a/d470996a77c1cc21fc04f56bf831e62a89a45d2708b57c10c2ea227623389c4a.vmx opID=030cbc40 user=root] Invalid transition requested (VM_STATE_INVALID_CONFIG -> VM_STATE_DELETING): Invalid state
2016-08-04T21:54:09.869Z info hostd[809C1B70] [Originator@6876 sub=Default opID=030cbc40 user=root] AdapterServer caught exception: vim.fault.InvalidPowerState
2016-08-04T21:54:09.869Z info hostd[809C1B70] [Originator@6876 sub=Vimsvc.TaskManager opID=030cbc40 user=root] Task Completed : haTask-2068-vim.ManagedEntity.destroy-368322 Status error
```

**Comment by hickeng** [comment]
It looks like the incorrect password is from vicadmin and probably not related to the VM invalid state issue.

**Comment by mdubya66** [comment]
assigning to @hickeng for initial triage

**Comment by mhagen-vmware** [comment]
Still being seen in CI:

[integration_test_logs_3713_1470506382.zip](https://github.com/vmware/vic/files/407048/integration_test_logs_3713_1470506382.zip)

**Comment by hickeng** [comment]
There are two portions to this:
- [ ] #1870 to address the portlayer shutdown logic
- [ ] why didn't the container shut down - this is blocked by the lack of collection of container debug logs by vicadmin (#1836 and the corresponding "incorrect username/password" issues)

**Comment by mdubya66** [comment]
This is a blocker to CI functioning correctly. Raising it's priority. @hmahmood will you take point on this and the other issues in this list?

#1878, #1870, #1822, #1823, #1109

**Comment by mdubya66** [comment]
This is a blocker to CI functioning correctly. Raising it's priority. @hmahmood will you take point on this and the other issues in this list?

#1878, #1870, #1822, #1823, #1109

**Comment by matthewavery** [comment]
Looks like both issues that were found to be tied to this one have been closed with merged fixes. I will be investigating whether this issue has been resolved now.

**Comment by mhagen-vmware** [comment]
[integration_logs_3947_140c4183042152f95cc42e790b5c7cfda58df935_1471027500.zip](https://github.com/vmware/vic/files/416015/integration_logs_3947_140c4183042152f95cc42e790b5c7cfda58df935_1471027500.zip)

**Comment by hickeng** [comment]
This should be addressed by checking for the illegal state reponse (text reason is "The operation is not allowed in the current state", but that's true if the VM is still powered on as well so we should attempt to get the actual state).
The fallback path in this scenario is to unregister the VM and then delete it's folder using FileManager.

**Comment by hickeng** [comment]
At this time I'm inclined towards the following:
1. add slight delay (which makes me cringe) in the containerVM shutdown path to avoid guestinfo updates colliding with power off. This should reduce the indcience of the issue.
2. add a check to the delete path that falls back to unregister/filemanager.remove
3. document that this can occur and that deleting the container is the solution.

**Comment by hmahmood** [comment]
One cause of this is active property collectors on the VM config during shutdown leading to invalid VM config in the aftermath. Container rm comes after stop, and exposes this invalid VM state. We do wait for the power state to change after we issue a kill to the container process, and that power state change wait relies on the property collector. So at least property collector is active during shutdown.

Currently the only other way is to poll for power state. However, @cgtexmex has a PR out for VM events, which can be utilized to wait for power state to go to off. That should go in shortly, and after that we can attempt to implement a solution for this.

**Comment by hickeng** [comment]
@hmahmood The event based logic should help but we cannot guarantee that no property collectors are registered when the VM shuts down so we will need the fallback logics as well.

**Comment by mhagen-vmware** [comment]
[integration_logs_4280_d6def22e628d9502782fc1bde453d3b2ae457fc3_1471962043.zip](https://github.com/vmware/vic/files/432700/integration_logs_4280_d6def22e628d9502782fc1bde453d3b2ae457fc3_1471962043.zip)

**Comment by anchal-agrawal** [comment]
Ran into this on a recent PR build.

[VCH-4385-5605-container-logs.zip](https://github.com/vmware/vic/files/437936/VCH-4385-5605-container-logs.zip)

**Comment by hmahmood** [comment]
@jzt is going to take care of the fall back logic here -- unregister followed by delete from disk.

**Comment by mhagen-vmware** [comment]
[container-logs.zip](https://github.com/vmware/vic/files/450310/container-logs.zip)

New longevity test fails consistently with this error.

**Comment by mhagen-vmware** [comment]
Reproduced on the latest code:
[container-logs.zip](https://github.com/vmware/vic/files/459498/container-logs.zip)

**Comment by mhagen-vmware** [comment]
This is currently a blocker for the longevity test.

**Comment by jzt** [comment]
This issue has a [PR](https://github.com/vmware/vic/issues/2230) out ready to be merged once  #2241 is determined. @hickeng is there a way we can expedite #2241 with @fdawg4l out this week?

**Comment by mhagen-vmware** [comment]
https://ci.vmware.run/vmware/vic/5126

[integration_logs_5126_86ffacd1c5da2ee4dcfe6aa4e6210ab7b43887ed.zip](https://github.com/vmware/vic/files/493587/integration_logs_5126_86ffacd1c5da2ee4dcfe6aa4e6210ab7b43887ed.zip)

**Comment by hickeng** [comment]
Upgrade tests hit this while trying to power a VM back on.

https://ci.vmware.run/vmware/vic/5693

**Comment by chengwang86** [comment]
@hickeng Could you pls put the logs here?

**Comment by emlin** [comment]
@chengwang86 you can get log address from the log above

**Comment by mdubya66** [comment]
Workaround: For Developer run docker create again, for VI Admin Un-register and re-register the VM in the vSphere UI.

**Comment by npakrasi** [comment]
Added to 0.7.0 known issues.

**Comment by npakrasi** [comment]
Added to 0.7.0 release notes known issues.

**Comment by mlh78750** [comment]
Since this has a workaround and noted in known issues, we're pushing this to .9

**Comment by tompscanlan** [comment]
I'm seeing something that may be the same thing in vic-machine-darwin version v1.1.0-rc5-9735-032392c.

```
 machine$ docker ps -a
CONTAINER ID        IMAGE                                   COMMAND             CREATED             STATUS              PORTS               NAMES
3b4cd076f362        harbor.home.local/tompscanlan/busybox   "ls"                3 minutes ago       Starting                                keen_fermat
d0e6f49207d6        harbor.home.local/tompscanlan/busybox   "/bin/sh"           5 minutes ago       Starting                                determined_yonath
6ba72a99a6c4        harbor.home.local/tompscanlan/busybox   "/bin/sh"           7 minutes ago       Starting                                infallible_almeida

 machine$  docker kill 6ba72a99a6c4 d0e6f49207d6 3b4cd076f362
Error response from daemon: Cannot kill container 6ba72a99a6c4: 6ba72a99a6c41a4eb2b183acaa4de26b9dea47c2288d6a1967f66f8a9b16127d is not running
Error response from daemon: Cannot kill container d0e6f49207d6: d0e6f49207d62b5eb4e3667caeff0508fb20f25723618040a6dc6a4f51a69471 is not running
Error response from daemon: Cannot kill container 3b4cd076f362: 3b4cd076f3624b4c8b2ed1444522621698593bb98c7f8530eb3b5c31a67c351e is not running

 machine$ docker rm 6ba72a99a6c4 d0e6f49207d6 3b4cd076f362
Error response from daemon: Server error from portlayer: [DELETE /containers/{id}][500] containerRemoveInternalServerError
Error response from daemon: Server error from portlayer: [DELETE /containers/{id}][500] containerRemoveInternalServerError
Error response from daemon: Server error from portlayer: [DELETE /containers/{id}][500] containerRemoveInternalServerError
```

[container-logs (7).zip](https://github.com/vmware/vic/files/936630/container-logs.7.zip)


vch was newly created using:
```

        vic-machine-darwin create \
            --target https://192.168.1.9/Goddard \
            --user administrator@home.local \
            --password 'VMware1!' \
            --name ${vch} \
            --tls-cname "${vch}.home.local" \
            --image-store esx-a-ssd  \
            --bridge-network bridge-${vch} \
            --compute-resource BasementLab \
            --thumbprint "29:03:72:8B:73:ED:D8:B6:D7:36:E8:EE:4F:1D:91:DE:9A:2C:3D:4A"  \
            --bridge-network-range 172.16.0.0/12  \
            --management-network Management \
            --management-network-gateway 0.0.0.0/0:192.168.1.254 \
            --management-network-ip 192.168.1.17/24  \
            --dns-server 192.168.1.1  \
            --public-network VMNet \
            --organization tscanlan   \
            --registry-ca harbor-ca.crt   \
            --volume-store "esx-a-ssd/${vch}-volumes:default"  \
            --timeout 10m

```

**Comment by hickeng** [comment]
@tompscanlan I think it's more likely to be #4577 - which can be hit from a couple of paths. #4613 is an example of the network serial connection failing (due to asymmetric routing setup in that case). There's also #4296.

**Comment by mhagen-vmware** [comment]
no longer seen

**Comment by jzt** [comment]
seen in 
[5-4-High-Availability.zip](https://github.com/vmware/vic/files/1011052/5-4-High-Availability.zip)

**Comment by mdubya66** [comment]
https://github.com/vmware/admiral/issues/102 was also filed against admiral

**Comment by jzt** [comment]
[5-4-High-Availability.zip](https://github.com/vmware/vic/files/1015407/5-4-High-Availability.zip)

**Comment by corrieb** [comment]
Note there are numerous issues open reporting a containerRemoveInternalServerError.
#5628
#5198
#5157
#4858

We need to get to the bottom of this.

**Comment by cgtexmex** [comment]
Might be fixed via #5198 or #5784, but a dupe of #5628
